packages/integration/src/bitbucket/core.ts

```diff
@@ -14,9 +14,84 @@
  * limitations under the License.
  */
 
+import fetch from 'cross-fetch';
 import parseGitUrl from 'git-url-parse';
 import { BitbucketIntegrationConfig } from './config';
 
+/**
+ * Given a URL pointing to a path on a provider, returns the default branch.
+ *
+ * @param url A URL pointing to a path
+ * @param config The relevant provider config
+ */
+export async function getBitbucketDefaultBranch(
+  url: string,
+  config: BitbucketIntegrationConfig,
+): Promise<string> {
+  const { name: repoName, owner: project, resource } = parseGitUrl(url);
+
+  const isHosted = resource === 'bitbucket.org';
+  const branchUrl = isHosted
+    ? `${config.apiBaseUrl}/repositories/${project}/${repoName}`
```

Comment:
The default branch is under `mainbranch.name`.
https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D#get